flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -198,10 +211,47 @@ else if (message instanceof SubmitJobAndWait) {
 					decorateMessage(new Status.Failure(new Exception(msg))), ActorRef.noSender());
 			}
 		}
+		else if (message instanceof AttachToJobAndWait) {
```

Comment:
This branch looks like sharing a lot of code with the branch above. I wonder if there's a way to generalize the code into a method?